sklearn/preprocessing/_encoders.py

```diff
@@ -208,14 +238,51 @@ class OneHotEncoder(_BaseEncoder):
     dtype : number type, default=np.float
         Desired dtype of output.
 
-    handle_unknown : {'error', 'ignore'}, default='error'
+    handle_unknown : {'error', 'ignore', 'auto'}, default='error'
         Whether to raise an error or ignore if an unknown categorical feature
         is present during transform (default is to raise). When this parameter
         is set to 'ignore' and an unknown category is encountered during
         transform, the resulting one-hot encoded columns for this feature
         will be all zeros. In the inverse transform, an unknown category
         will be denoted as None.
 
+        When this parameter is set to 'auto' and an unknown category is
+        encountered in transform:
+
+            1. If there was no infrequent category during training, the
+            resulting one-hot encoded columns for this feature will be all
+            zeros. In the inverse transform, an unknown category will be
+            denoted as None.
+
+            2. If there is an infrequent category during training, the unknown
+            category will be considered infrequent. In the inverse transform,
+            an unknown category will be the most frequent infrequent category
```

Comment:
frequent unfrequent -> Ouch my brain :)

the infrequent category with the highest frequency?

```suggestion
            an unknown category will be the most frequent infrequent category.
```